src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -58,23 +61,52 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 	}
 
 	/**
-	 * Trims leading and trailing whitespace, converts empty strings to null, and then
-         * checks that a string is either null or has at least one non-whitespace character
-         * and conforms to the specified regular expression.
-         * If optional is true, null is a valid output value; if false, null will throw an error.
-         * If replace is not null, it is used for a replaceAll operation, and the
+	 * Returns a new immutable list with nulls and duplicates removed
```

Comment:
I believe we had a similar discussion during the bulk import work.

My philosophy is to fix easily-corrected errors (DWIM) instead of rejecting the input -- hence removing nulls / dupes.

Reply:
Correcting easy errors is one interpretation; another is that it's hiding bugs that could wind up causing issues elsewhere and enabling sloppy coding practice upstream.

That being said, it's probably not worth the energy we'd spend debating it so, FINE, whateverrrrrrr

:+1: